### ISSUE TITLE
docker network create  error

### ISSUE THREAD

**Issue by Dreampie** [docker/docker#18770]
```
docker pull swarm
vim /etc/default/docker
DOCKER_OPTS=“-H 0.0.0.0:2375 –H unix:///var/run/docker.sock”
systemctl daemon-reload
systemctl restart docker
ps auxwww | grep docker

sudo docker run --rm swarm create
85e254bb7067a2e07f9dd4fedc544f94

docker run -d swarm join --addr=192.168.60.43:2375 token://85e254bb7067a2e07f9dd4fedc544f94

docker run -d -p 2376:2375 swarm manage token://85e254bb7067a2e07f9dd4fedc544f94
```

<font color="red">Error:</font>
```java
docker network create --driver overlay swarm-network
Error response from daemon: failed to parse pool request for address space "GlobalDefault" pool "" subpool "": cannot find address space GlobalDefault (most likely the backing datastore is not configured)
```

```
docker -H tcp://127.0.0.1:2376 info
Containers: 19
Images: 17
Role: primary
Strategy: spread
Filters: health, port, dependency, affinity, constraint
Nodes: 2
 localhost.localdomain: 192.168.60.41:2375
  └ Status: Healthy
  └ Containers: 12
  └ Reserved CPUs: 0 / 1
  └ Reserved Memory: 0 B / 1.018 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.10.0-229.20.1.el7.x86_64, operatingsystem=CentOS Linux 7 (Core), storagedriver=devicemapper
 localhost.localdomain: 192.168.60.43:2375
  └ Status: Healthy
  └ Containers: 7
  └ Reserved CPUs: 0 / 1
  └ Reserved Memory: 0 B / 1.018 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.10.0-327.3.1.el7.x86_64, operatingsystem=CentOS Linux 7 (Core), storagedriver=devicemapper
CPUs: 2
Total Memory: 2.037 GiB
Name: 9e8b36f2cdbe
```

```
docker version && docker info && uname -a
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:25:01 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:25:01 UTC 2015
 OS/Arch:      linux/amd64
Containers: 12
Images: 119
Server Version: 1.9.1
Storage Driver: devicemapper
 Pool Name: docker-253:0-33993639-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 107.4 GB
 Backing Filesystem:
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 2.977 GB
 Data Space Total: 107.4 GB
 Data Space Available: 7.017 GB
 Metadata Space Used: 5.849 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.142 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.93-RHEL7 (2015-01-28)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.10.0-229.20.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
CPUs: 1
Total Memory: 993.3 MiB
Name: localhost.localdomain
ID: R2NX:ZX6S:PQIE:R65Y:X62A:ACXU:MALH:RXHK:BRF5:6RMU:NJLG:TMPL
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
Linux localhost.localdomain 3.10.0-229.20.1.el7.x86_64 #1 SMP Tue Nov 3 19:10:07 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by thaJeztah** [comment]
Haven't looked in-depth yet, but note that;

- the `/etc/default/docker` file is NOT used for systems running systemd. to configure a daemon running on a system with systemd, see this section: https://docs.docker.com/engine/articles/systemd/#custom-docker-daemon-options
- overlay networking requires kernel 3.16 or higher. You'll also have to set up a key-value store, see https://docs.docker.com/engine/userguide/networking/get-started-overlay/

Looking at the output provided, at least you don't have kernel 3.16 or higher, and from the error-message, don't have a key-value store set up. I suggest to read the "get stated with multi-host networking" section first https://docs.docker.com/engine/userguide/networking/get-started-overlay/

I'm going to close this issue, because this looks like it's the result of incorrect configuration and requirements, but feel free to comment after I've closed this issue

**Comment by Dreampie** [comment]
the /etc/default/docker file is NOT used for systems running systemd?
/etc/default/docker:
"--insecure-registry 192.168.60.41:5000 -H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock"  in use.
```
ps auxwww | grep docker
root      9665  0.0  0.0 112656   932 pts/0    S+   20:58   0:00 grep --color=auto docker
root     22588  0.7  2.9 910656 30128 ?        Ssl  12月18  26:58 /usr/bin/docker daemon -H fd:// --insecure-registry 192.168.60.41:5000 -H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock
```
overlay networking requires kernel 3.16 or higher?
"kernelversion=3.10.0-327.3.1.el7.x86_64" in use.

```
docker -H tcp://127.0.0.1:2376 info
Containers: 19
Images: 17
Role: primary
Strategy: spread
Filters: health, port, dependency, affinity, constraint
Nodes: 2
 localhost.localdomain: 192.168.60.41:2375
  └ Status: Healthy
  └ Containers: 12
  └ Reserved CPUs: 0 / 1
  └ Reserved Memory: 0 B / 1.018 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.10.0-229.20.1.el7.x86_64, operatingsystem=CentOS Linux 7 (Core), storagedriver=devicemapper
 localhost.localdomain: 192.168.60.43:2375
  └ Status: Healthy
  └ Containers: 7
  └ Reserved CPUs: 0 / 1
  └ Reserved Memory: 0 B / 1.018 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.10.0-327.3.1.el7.x86_64, operatingsystem=CentOS Linux 7 (Core), storagedriver=devicemapper
CPUs: 2
Total Memory: 2.037 GiB
Name: 9e8b36f2cdbe
```
don't have a key-value store set up?  my docker swarm always run in centos virtualbox, I can't use docker-machine to create the virtualbox,so I'm not run consul in docker container.

How to resolve this problem?

**Comment by mavenugo** [comment]
@Dreampie For systemd configuration, please refer to https://docs.docker.com/engine/articles/systemd/#custom-docker-daemon-options
We have resolved the kernel < 3.16 issue via https://github.com/docker/docker/pull/18775. Once this is merged in master, you can use it with kernel 3.10 as well.

**Comment by Dreampie** [comment]
thank you. I resolve this. but got other error: 

```
vim /etc/default/docker
DOCKER_OPTS="--insecure-registry 192.168.60.41:5000 -H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock --cluster-store=consul://192.168.60.41:8500 --cluster-advertise=ens18:2376"
```

```
docker run -d -p 8888:8888 -v /var/log --name config-server --net=swarm-network 192.168.60.41:5000/config-server:1.0-SNAPSHOT.1
d02ee2ee33d3755ab826b668e3d7cf4a6f4a8852c17ce006887f6c704f4ba050
Error response from daemon: Cannot start container d02ee2ee33d3755ab826b668e3d7cf4a6f4a8852c17ce006887f6c704f4ba050: subnet sandbox join failed for "10.0.0.0/24": vxlan interface creation failed for subnet "10.0.0.0/24": failed in prefunc: failed to set namespace on link "vxlan4db5a43": invalid argument
```
```
docker network ls
NETWORK ID          NAME                DRIVER
fd8e646cbdad        swarm-network       overlay
588d4b1417c3        bridge              bridge
c2fe2bc6ade3        none                null
cc30063d2ce1        host                host
```

**Comment by mavenugo** [comment]
@Dreampie yes. thats the need for the kernel 3.16 upgrade. But once https://github.com/docker/docker/pull/18775 is merged, `invalid argument` issue that you see will be resolved as well.

**Comment by Dreampie** [comment]
error change : 
```
Error response from daemon: Cannot start container 58ac56418f721d09f2a009c327316a36b40fd85906c7511a087f923fe65d0b75: subnet sandbox join failed for "10.0.0.0/24": error creating vxlan interface: file exists
```

**Comment by mavenugo** [comment]
@Dreampie could you please reboot the VM/machine once and try again ?

**Comment by Dreampie** [comment]
yes，if reboot, overlay network will lost?  I got other error:
My mongo server container and Service A container all at 192.168.60.41, Service A connection 192.168.60.41:21017(mongo), get error:
![image](https://cloud.githubusercontent.com/assets/3667080/11929806/20552738-a819-11e5-9d0e-b81555cf7098.png)

```
Caused by: feign.FeignException: status 500 reading UserService#findAll(); content:
{"timestamp":1450696995019,"status":500,"error":"Internal Server Error","exception":"org.springframework.dao.DataAccessResourceFailureException","message":"Request processing failed; nested exception is org.springframework.dao.DataAccessResourceFailureException: Timed out after 10000 ms while waiting to connect. Client view of cluster state is {type=Unknown, servers=[{address=192.168.60.41:27017, type=Unknown, state=Connecting, exception={com.mongodb.MongoException$Network: Exception opening the socket}, caused by {java.net.SocketTimeoutException: connect timed out}}]; nested exception is com.mongodb.MongoTimeoutException: Timed out after 10000 ms while waiting to connect. Client view of cluster state is {type=Unknown, servers=[{address=192.168.60.41:27017, type=Unknown, state=Connecting, exception={com.mongodb.MongoException$Network: Exception opening the socket}, caused by {java.net.SocketTimeoutException: connect timed out}}]","path":"/users"}
	at feign.FeignException.errorStatus(FeignException.java:52)
	at feign.codec.ErrorDecoder$Default.decode(ErrorDecoder.java:81)
	at feign.SynchronousMethodHandler.executeAndDecode(SynchronousMethodHandler.java:121)
	at feign.SynchronousMethodHandler.invoke(SynchronousMethodHandler.java:71)
	at feign.ReflectiveFeign$FeignInvocationHandler.invoke(ReflectiveFeign.java:94)
	at com.sun.proxy.$Proxy92.findAll(Unknown Source)
	at cn.dreampie.api.user.controller.UserController.findAll(UserController.java:23)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:221)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:137)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:110)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandleMethod(RequestMappingHandlerAdapter.java:776)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:705)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:85)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:959)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:893)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:966)
	... 48 common frames omitted

```

but other container(Service A) at 192.168.60.43 connection 192.168.60.41:27017 is ok.

![image](https://cloud.githubusercontent.com/assets/3667080/11929800/14fc8cdc-a819-11e5-9e33-a146b908fb43.png)

```
docker run --rm swarm list token://d771f62550d899aa9f39352eec997793
192.168.60.41:2375
192.168.60.43:2375
```
![image](https://cloud.githubusercontent.com/assets/3667080/11929820/42757e62-a819-11e5-8be4-591dde8f9c15.png)
